### PR TITLE
Added acr build image config 

### PR DESCRIPTION
**Pull request by linusnorton**
Added acr build image config to build individual images for ccd and probate configuration files

### CODE REVIEW

File: acr-build.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+version: v1.1.0
+steps:
+  - id: ccd
+    build: >
+      -t {{.Run.Registry}}/ccd/logstash/ccd
+      --target ccd
+      --build-arg conf=ccd_logstash.conf.in
+      .
+    when:
+      - pull-ccd
```
Comment (linusnorton) [inline acr-build.yml:10]: @timja as I've removed the pull step should this now be `["-"]` or is there a different value?

Comment (timja) [reply acr-build.yml:10]: just remove the when, see the simpler example I sent you https://github.com/hmcts/cnp-jenkins-docker/blob/master/jenkins-agent/acr-agents-build-task.yaml

File: acr-build.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+version: v1.1.0
+steps:
+  - id: ccd
+    build: >
+      -t {{.Run.Registry}}/ccd/logstash/ccd
```
Comment (timja) [inline acr-build.yml:5]:
```suggestion
      -t {{.Run.Registry}}/ccd/logstash:ccd-{{.Run.ID}} -t {{.Run.Registry}}/ccd/logstash:ccd-latest -t {{.Run.Registry}}/ccd/logstash:latest
```

File: acr-build.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      --target ccd
```
Comment (timja) [inline acr-build.yml:6]: what's this do?

File: acr-build.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      --build-arg conf=ccd_logstash.conf.in
+      .
+    when:
+      - pull-ccd
+    keep: true
+
+  - id: probate
+    build: >
+      -t {{.Run.Registry}}/ccd/logstash/probate
```
Comment (timja) [inline acr-build.yml:15]:
```suggestion
      -t {{.Run.Registry}}/ccd/logstash:probate-{{.Run.ID}} -t {{.Run.Registry}}/ccd/logstash:probate-latest
```

File: bin/acr-task-creation.sh [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+export ACR_NAME=hmctspublic
+export GIT_PAT=$(az keyvault secret show --vault-name infra-vault-prod --name hmcts-github-apikey --query value -o tsv)
+
+az acr task create \
+  --registry $ACR_NAME \
+  --name ccd-logstash \
+  --context https://github.com/hmcts/ccd-logstash.git \
+  --file acr-build.yaml \
```
Comment (timja) [inline bin/acr-task-creation.sh:10]: yaml or yml? the file is using yml currently
```suggestion
  --file acr-build.yml \
```

File: acr-build.yml (hunk 3 above)
Comment (linusnorton) [reply acr-build.yml:6]: copy pasta

File: acr-build.yml (hunk 1 above)
Comment (timja) [inline acr-build.yml:10]:
```suggestion
```

File: bin/acr-task-creation.sh (hunk 5 above)
Comment (linusnorton) [reply bin/acr-task-creation.sh:10]: :facepalm:

File: acr-build.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    when:
+      - pull-probate
+    keep: true
+
+  - id: push-images
+    push:
+      - "{{.Run.Registry}}/ccd/logstash/ccd:latest"
+      - "{{.Run.Registry}}/ccd/logstash/probate:latest"
+    when:
+      - runtime
```
Comment (timja) [inline acr-build.yml:26]:
```suggestion
    when:
      - probate
      - ccd
```

File: acr-build.yml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    keep: true
+
+  - id: probate
+    build: >
+      -t {{.Run.Registry}}/ccd/logstash/probate
+      --build-arg conf=ccd_logstash.conf.in
+      --target probate
+      .
+    when:
+      - pull-probate
```
Comment (timja) [inline acr-build.yml:18]:
```suggestion
```

File: acr-build.yml [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+      .
+    when:
+      - pull-ccd
+    keep: true
+
+  - id: probate
+    build: >
+      -t {{.Run.Registry}}/ccd/logstash/probate
+      --build-arg conf=ccd_logstash.conf.in
+      --target probate
```
Comment (timja) [inline acr-build.yml:15]:
```suggestion
```

File: acr-build.yml [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    build: >
+      -t {{.Run.Registry}}/ccd/logstash:probate-{{.Run.ID}} -t {{.Run.Registry}}/ccd/logstash:probate-latest
+      --build-arg conf=probate_logstash.conf.in
+      .
+    keep: true
+
+  - id: push-images
+    push:
+      - "{{.Run.Registry}}/ccd/logstash/ccd:latest"
+      - "{{.Run.Registry}}/ccd/logstash/probate:latest"
```
Comment (timja) [inline acr-build.yml:20]: this should add all the required images on push
```suggestion
      - "{{.Run.Registry}}/ccd/logstash:latest"
      - "{{.Run.Registry}}/ccd/logstash:ccd-latest"
      - "{{.Run.Registry}}/ccd/logstash:ccd-{{.Run.ID}}"
      - "{{.Run.Registry}}/ccd/logstash:probate-latest"
      - "{{.Run.Registry}}/ccd/logstash:probate-{{.Run.ID}}"
```

**Pull request merged**